smarts/core/coordinates.py

```diff
@@ -65,11 +66,35 @@ def equal_if_defined(self, length: float, width: float, height: float) -> bool:
         )
 
 
+_shapely_points = {}
+
+
 class Point(NamedTuple):
     x: float
     y: float
     z: Optional[float] = 0
 
+    @property
+    def as_shapely(self) -> SPoint:
+        # Shapley Point construction is expensive!
+        # Note that before python3.8, @cached_property was not thread safe,
+        # nor can it be used in a NamedTuple (which doesn't have a __dict__).
+        # (Points can be used by multi-threaded client code, even when
+        # SMARTS is still single-threaded, so we want to be safe here.)
+        # So we use the private global _shapely_points as a cache instead.
+        # Here we are relying on CPython's implementation of dict
+        # to be thread-safe.
```

Comment:
Just checking -- are we sure we don't need to worry about this for our other uses of `@cached_property`?

Reply:
> Just checking -- are we sure we don't need to worry about this for our other uses of `@cached_property`?

lol, no!

I am fairly confident (above 95%) that the vast majority of our uses of `@cached_property` are ok, as SMARTS is only supposed to be run with a single thread anyway.  However,  internal uses that come via the ROS wrapper can possibly made on other threads, so I did audit those uses (as well as uses in an alternative `RoadMap` implementation that has background processes -- all uses there were converted to `@threaded_cached_property`).  

This is partly why I added the comment above, in order to remind myself and others every time I look at this code, in case we encounter some weirdness down the road that might be related to an unsafe use of `@cached_property`.  

But while at this point, I am still not 100% confident that we have managed this appropriately, I do thin the above (including the comment) is an improvement over what we had originally, so I'm willing to go ahead and merge the PR now.

Reply:
I say go ahead when you have the chance.